src/app/visit-request/components/modals/visit-request-management-modal/visit-request-management-modal.component.ts

```diff
@@ -0,0 +1,115 @@
+import { Component } from '@angular/core';
+import { NgbActiveModal } from '@ng-bootstrap/ng-bootstrap';
+import { AgentModel } from '../../../../common/models/agent-model';
+import { AgentService } from '../../../../agent/services/agent.service';
+import { VisitRequestService } from '../../../services/visit-request.service';
+import { ToastService } from '../../../../common/services/toast-service/toast.service';
+import { ToastClassEnum } from '../../../../common/enums/toast-class-enum';
+import { RealEstateHeader } from '../../../../common/models/real-estate-management-model';
+import { RealEstateManagementApiService } from '../../../../realestate/services/real-estate-management-api.service';
+import { visitRequestForm } from '../../../../common/services/form/form.service';
+import { VisitRequestModel } from '../../../../common/models/visit-request-model';
+import { visitRequestAvailabilityDto } from '../../../../common/dto/visit-request-availability-dto';
+
+@Component({
+  selector: 'app-visit-request-management-modal',
+  templateUrl: './visit-request-management-modal.component.html',
+  styleUrls: ['./visit-request-management-modal.component.css']
+})
+export class VisitRequestManagementModalComponent {
+  form = visitRequestForm;
+  minDate: string;
+  agents: AgentModel[];
+  realestates: RealEstateHeader[];
+  times: string[] = [];
+  endTimes: string[] = [];
+  filteredEndTimes: string[] = [];
+  startTime: string;
+  endTime: string;
+  date: string;
+  visitRequestModel: VisitRequestModel
+  isAvailable: boolean = false
+  realEstateId: number
+  errorMessage: string
+
+  constructor(
+    public activeModal: NgbActiveModal,
+    private agentService: AgentService,
+    private realEstateManagementApiService: RealEstateManagementApiService,
+    private visitRequestService: VisitRequestService,
+    private toastService: ToastService,
+  ) {
+
+    const today = new Date();
+    this.minDate = today.toISOString().split('T')[0];
+
+    for (let hour = 9; hour < 20; hour++) {
+      const timeString = hour < 10 ? `0${hour}:00` : `${hour}:00`;
+      this.times.push(timeString);
+      this.endTimes.push(timeString);
+    }
+    this.endTimes.push('20:00');
+  }
+
+  ngOnInit(): void {
+    this.form.controls.realEstateId.setValue(this.realEstateId)
+    this.agentService.getAllAgentData().subscribe(response => {
```

Comment:
Move getAllAgentData and getAllRealEstates to a method called: loadAgentAndRealEstateData()
Don't forget to call this.loadAgentAndRealEstateData() in ngOnInit.